src/use-utility-connect.js

```diff
@@ -16,32 +19,59 @@ const getConfigError = errors => {
 };
 
 export const useUtilityConnect = () => {
-  const [error, setError] = useState();
-  const [loading, setLoading] = useState(false);
+  const [utilityConnectError, setUtilityConnectError] = useState();
+  const [widgetOpening, setWidgetOpening] = useState(false);
   const [factory, setFactory] = useState();
 
   const [openOnScriptLoad, setOpenOnScriptLoad] = useState(false);
   const [savedOpenParams, setSavedOpenParams] = useState();
+  const pollForFactoryTimeout = useRef(null);
+
+  const pollingCount = useRef(0);
 
   const [scriptLoading, scriptError] = useScript({
     src,
     checkForExisting: true,
   });
 
-  useEffect(() => {
-    if (scriptLoading || scriptError) return;
-
+  const pollForFactory = useCallback(() => {
     const { _ArcadiaUtilityConnect } = window;
-    if (!_ArcadiaUtilityConnect) {
-      setError(initializeError);
+    if (_ArcadiaUtilityConnect) {
+      setFactory(_ArcadiaUtilityConnect);
+      return;
+    }
+
+    if (pollingCount.current === MAX_POLL_NUMBER) {
+      pollingCount.current = 0;
```

Comment:
why do we have to reset this to 0 before returning the error?

Reply:
We don't _have_ to, but I figured it's good practice just in case we add a retry here some day - it would be very easy to miss.